### ISSUE TITLE
the Role of Adding Gaussian Noise in the text classifier training

### ISSUE THREAD

**Issue by Adasunnylily** [jmiemirza/TAP#1]
hi, great work!
 I am confused about the use of  Gaussian noise to the data  when training the text classifier and how it will effect the result. Here is the relevant code snippet: 
def train_txt_clas(self, criteria):
        noise_std = 0.1
        noise = torch.randn(self.txt_features_for_text_cls.shape) * noise_std
        txt_feas = self.txt_features_for_text_cls
        txt_label = self.labels_for_text_cls
        feas = (self.text_adapter(txt_feas.to(torch.float32) + noise.cuda()))
        loss = criteria(feas, txt_label)
        return loss

**Comment by jmiemirza** [comment]
Hi, 

Thanks for your interest. Since language is very sparse (as compared to images), so in order to avoid overfitting to the text dataset, we add Gaussian Noise to the text features obtained from the text-encoder. You can think of it as a simple regulariser. 

Best,
Mirza.
